File: web/lib/supabase/types.gen.ts
export type Json =
  | string
  | number
  | boolean
  | null
  | { [key: string]: Json | undefined }
  | Json[]

export type Database = {
  graphql_public: {
    Tables: {
      [_ in never]: never
    }
    Views: {
      [_ in never]: never
    }
    Functions: {
      graphql: {
        Args: {
          operationName?: string
          extensions?: Json
          variables?: Json
          query?: string
        }
        Returns: Json
      }
    }
    Enums: {
      [_ in never]: never
    }
    CompositeTypes: {
      [_ in never]: never
    }
  }
  public: {
    Tables: {
      clients: {
        Row: {
          created_at: string
          display: string
          environment_id: string
          id: string
        }
        Insert: {
          created_at?: string
          display?: string
          environment_id: string
          id?: string
        }
        Update: {
          created_at?: string
          display?: string
          environment_id?: string
          id?: string
        }
        Relationships: [
          {
            foreignKeyName: "clients_environment_id_fkey"
            columns: ["environment_id"]
            isOneToOne: false
            referencedRelation: "environments"
            referencedColumns: ["id"]
          },
        ]
      }
      clients_secrets: {
        Row: {
          client_id: string
          created_at: string
          hash: string
          id: string
        }
        Insert: {
          client_id: string
          created_at?: string
          hash: string
          id?: string
        }
        Update: {
          client_id?: string
          created_at?: string
          hash?: string
          id?: string
        }
        Relationships: [
          {
            foreignKeyName: "clients_secrets_client_id_fkey"
            columns: ["client_id"]
            isOneToOne: false
            referencedRelation: "clients"
            referencedColumns: ["id"]
          },
        ]
      }
      environments: {
        Row: {
          created_at: string
          display: string
          id: string
          project_id: string
        }
        Insert: {
          created_at?: string
          display?: string
          id?: string
          project_id: string
        }
        Update: {
          created_at?: string
          display?: string
          id?: string
          project_id?: string
        }
        Relationships: [
          {
            foreignKeyName: "environments_project_id_fkey"
            columns: ["project_id"]
            isOneToOne: false
            referencedRelation: "projects"
            referencedColumns: ["id"]
          },
        ]
      }
      projects: {
        Row: {
          display: string
          id: string
        }
        Insert: {
          display?: string
          id?: string
        }
        Update: {
          display?: string
          id?: string
        }
        Relationships: []
      }
      secrets: {
        Row: {
          created_at: string
          environment_id: string
          id: string
          value: string
          variable_id: string
        }
        Insert: {
          created_at?: string
          environment_id: string
          id?: string
          value?: string
          variable_id: string
        }
        Update: {
          created_at?: string
          environment_id?: string
          id?: string
          value?: string
          variable_id?: string
        }
        Relationships: [
          {
            foreignKeyName: "secrets_environment_id_fkey"
            columns: ["environment_id"]
            isOneToOne: false
            referencedRelation: "environments"
            referencedColumns: ["id"]
          },
          {
            foreignKeyName: "secrets_variable_id_fkey"
            columns: ["variable_id"]
            isOneToOne: false
            referencedRelation: "variables"
            referencedColumns: ["id"]
          },
        ]
      }
      variables: {
        Row: {
          description: string
          generator_data: Json
          generator_type: Database["public"]["Enums"]["generator"]
          id: string
          key: string
          project_id: string
        }
        Insert: {
          description?: string
          generator_data: Json
          generator_type: Database["public"]["Enums"]["generator"]
          id?: string
          key: string
          project_id: string
        }
        Update: {
          description?: string
          generator_data?: Json
          generator_type?: Database["public"]["Enums"]["generator"]
          id?: string
          key?: string
          project_id?: string
        }
        Relationships: [
          {
            foreignKeyName: "variables_project_id_fkey"
            columns: ["project_id"]
            isOneToOne: false
            referencedRelation: "projects"
            referencedColumns: ["id"]
          },
        ]
      }
    }
    Views: {
      [_ in never]: never
    }
    Functions: {
      create_client_and_secret: {
        Args: { p_display: string; p_env_id: string }
        Returns: {
          client_id: string
          secret_id: string
          secret: string
        }[]
      }
      create_client_secret: {
        Args: { p_secret: string; p_client_id: string }
        Returns: string
      }
      get_default_secret: {
        Args: { variable_id: string }
        Returns: string
      }
      is_admin_client: {
        Args: Record<PropertyKey, never>
        Returns: boolean
      }
      is_uuid: {
        Args: { s: string }
        Returns: boolean
      }
      is_valid_generator_data: {
        Args: { gdata: Json; gtype: Database["public"]["Enums"]["generator"] }
        Returns: boolean
      }
      verify_client_secret: {
        Args: { p_secret_id: string; p_secret: string }
        Returns: boolean
      }
    }
    Enums: {
      generator: "STATIC" | "RANDOM"
    }
    CompositeTypes: {
      [_ in never]: never
    }
  }
}

type DatabaseWithoutInternals = Omit<Database, "__InternalSupabase">

type DefaultSchema = DatabaseWithoutInternals[Extract<keyof Database, "public">]

export type Tables<
  DefaultSchemaTableNameOrOptions extends
    | keyof (DefaultSchema["Tables"] & DefaultSchema["Views"])
    | { schema: keyof DatabaseWithoutInternals },
  TableName extends DefaultSchemaTableNameOrOptions extends {
    schema: keyof DatabaseWithoutInternals
  }
    ? keyof (DatabaseWithoutInternals[DefaultSchemaTableNameOrOptions["schema"]]["Tables"] &
        DatabaseWithoutInternals[DefaultSchemaTableNameOrOptions["schema"]]["Views"])
    : never = never,
> = DefaultSchemaTableNameOrOptions extends {
  schema: keyof DatabaseWithoutInternals
}
  ? (DatabaseWithoutInternals[DefaultSchemaTableNameOrOptions["schema"]]["Tables"] &
      DatabaseWithoutInternals[DefaultSchemaTableNameOrOptions["schema"]]["Views"])[TableName] extends {
      Row: infer R
    }
    ? R
    : never
  : DefaultSchemaTableNameOrOptions extends keyof (DefaultSchema["Tables"] &
        DefaultSchema["Views"])
    ? (DefaultSchema["Tables"] &
        DefaultSchema["Views"])[DefaultSchemaTableNameOrOptions] extends {
        Row: infer R
      }
      ? R
      : never
    : never

export type TablesInsert<
  DefaultSchemaTableNameOrOptions extends
    | keyof DefaultSchema["Tables"]
    | { schema: keyof DatabaseWithoutInternals },
  TableName extends DefaultSchemaTableNameOrOptions extends {
    schema: keyof DatabaseWithoutInternals
  }
    ? keyof DatabaseWithoutInternals[DefaultSchemaTableNameOrOptions["schema"]]["Tables"]
    : never = never,
> = DefaultSchemaTableNameOrOptions extends {
  schema: keyof DatabaseWithoutInternals
}
  ? DatabaseWithoutInternals[DefaultSchemaTableNameOrOptions["schema"]]["Tables"][TableName] extends {
      Insert: infer I
    }
    ? I
    : never
  : DefaultSchemaTableNameOrOptions extends keyof DefaultSchema["Tables"]
    ? DefaultSchema["Tables"][DefaultSchemaTableNameOrOptions] extends {
        Insert: infer I
      }
      ? I
      : never
    : never

export type TablesUpdate<
  DefaultSchemaTableNameOrOptions extends
    | keyof DefaultSchema["Tables"]
    | { schema: keyof DatabaseWithoutInternals },
  TableName extends DefaultSchemaTableNameOrOptions extends {
    schema: keyof DatabaseWithoutInternals
  }
    ? keyof DatabaseWithoutInternals[DefaultSchemaTableNameOrOptions["schema"]]["Tables"]
    : never = never,
> = DefaultSchemaTableNameOrOptions extends {
  schema: keyof DatabaseWithoutInternals
}
  ? DatabaseWithoutInternals[DefaultSchemaTableNameOrOptions["schema"]]["Tables"][TableName] extends {
      Update: infer U
    }
    ? U
    : never
  : DefaultSchemaTableNameOrOptions extends keyof DefaultSchema["Tables"]
    ? DefaultSchema["Tables"][DefaultSchemaTableNameOrOptions] extends {
        Update: infer U
      }
      ? U
      : never
    : never

export type Enums<
  DefaultSchemaEnumNameOrOptions extends
    | keyof DefaultSchema["Enums"]
    | { schema: keyof DatabaseWithoutInternals },
  EnumName extends DefaultSchemaEnumNameOrOptions extends {
    schema: keyof DatabaseWithoutInternals
  }
    ? keyof DatabaseWithoutInternals[DefaultSchemaEnumNameOrOptions["schema"]]["Enums"]
    : never = never,
> = DefaultSchemaEnumNameOrOptions extends {
  schema: keyof DatabaseWithoutInternals
}
  ? DatabaseWithoutInternals[DefaultSchemaEnumNameOrOptions["schema"]]["Enums"][EnumName]
  : DefaultSchemaEnumNameOrOptions extends keyof DefaultSchema["Enums"]
    ? DefaultSchema["Enums"][DefaultSchemaEnumNameOrOptions]
    : never

export type CompositeTypes<
  PublicCompositeTypeNameOrOptions extends
    | keyof DefaultSchema["CompositeTypes"]
    | { schema: keyof DatabaseWithoutInternals },
  CompositeTypeName extends PublicCompositeTypeNameOrOptions extends {
    schema: keyof DatabaseWithoutInternals
  }
    ? keyof DatabaseWithoutInternals[PublicCompositeTypeNameOrOptions["schema"]]["CompositeTypes"]
    : never = never,
> = PublicCompositeTypeNameOrOptions extends {
  schema: keyof DatabaseWithoutInternals
}
  ? DatabaseWithoutInternals[PublicCompositeTypeNameOrOptions["schema"]]["CompositeTypes"][CompositeTypeName]
  : PublicCompositeTypeNameOrOptions extends keyof DefaultSchema["CompositeTypes"]
    ? DefaultSchema["CompositeTypes"][PublicCompositeTypeNameOrOptions]
    : never

export const Constants = {
  graphql_public: {
    Enums: {},
  },
  public: {
    Enums: {
      generator: ["STATIC", "RANDOM"],
    },
  },
} as const

File: go/pkg/supabase/database/types.gen.go
package database

type PublicProjectsSelect struct {
  Display string `json:"display"`
  Id      string `json:"id"`
}

type PublicProjectsInsert struct {
  Display *string `json:"display"`
  Id      *string `json:"id"`
}

type PublicProjectsUpdate struct {
  Display *string `json:"display"`
  Id      *string `json:"id"`
}

type PublicVariablesSelect struct {
  Description   string      `json:"description"`
  GeneratorData interface{} `json:"generator_data"`
  GeneratorType string      `json:"generator_type"`
  Id            string      `json:"id"`
  Key           string      `json:"key"`
  ProjectId     string      `json:"project_id"`
}

type PublicVariablesInsert struct {
  Description   *string     `json:"description"`
  GeneratorData interface{} `json:"generator_data"`
  GeneratorType string      `json:"generator_type"`
  Id            *string     `json:"id"`
  Key           string      `json:"key"`
  ProjectId     string      `json:"project_id"`
}

type PublicVariablesUpdate struct {
  Description   *string     `json:"description"`
  GeneratorData interface{} `json:"generator_data"`
  GeneratorType *string     `json:"generator_type"`
  Id            *string     `json:"id"`
  Key           *string     `json:"key"`
  ProjectId     *string     `json:"project_id"`
}

type PublicClientsSelect struct {
  CreatedAt     string `json:"created_at"`
  Display       string `json:"display"`
  EnvironmentId string `json:"environment_id"`
  Id            string `json:"id"`
}

type PublicClientsInsert struct {
  CreatedAt     *string `json:"created_at"`
  Display       *string `json:"display"`
  EnvironmentId string  `json:"environment_id"`
  Id            *string `json:"id"`
}

type PublicClientsUpdate struct {
  CreatedAt     *string `json:"created_at"`
  Display       *string `json:"display"`
  EnvironmentId *string `json:"environment_id"`
  Id            *string `json:"id"`
}

type PublicClientsSecretsSelect struct {
  ClientId  string `json:"client_id"`
  CreatedAt string `json:"created_at"`
  Hash      string `json:"hash"`
  Id        string `json:"id"`
}

type PublicClientsSecretsInsert struct {
  ClientId  string  `json:"client_id"`
  CreatedAt *string `json:"created_at"`
  Hash      string  `json:"hash"`
  Id        *string `json:"id"`
}

type PublicClientsSecretsUpdate struct {
  ClientId  *string `json:"client_id"`
  CreatedAt *string `json:"created_at"`
  Hash      *string `json:"hash"`
  Id        *string `json:"id"`
}

type PublicEnvironmentsSelect struct {
  CreatedAt string `json:"created_at"`
  Display   string `json:"display"`
  Id        string `json:"id"`
  ProjectId string `json:"project_id"`
}

type PublicEnvironmentsInsert struct {
  CreatedAt *string `json:"created_at"`
  Display   *string `json:"display"`
  Id        *string `json:"id"`
  ProjectId string  `json:"project_id"`
}

type PublicEnvironmentsUpdate struct {
  CreatedAt *string `json:"created_at"`
  Display   *string `json:"display"`
  Id        *string `json:"id"`
  ProjectId *string `json:"project_id"`
}

type PublicSecretsSelect struct {
  CreatedAt     string `json:"created_at"`
  EnvironmentId string `json:"environment_id"`
  Id            string `json:"id"`
  Value         string `json:"value"`
  VariableId    string `json:"variable_id"`
}

type PublicSecretsInsert struct {
  CreatedAt     *string `json:"created_at"`
  EnvironmentId string  `json:"environment_id"`
  Id            *string `json:"id"`
  Value         *string `json:"value"`
  VariableId    string  `json:"variable_id"`
}

type PublicSecretsUpdate struct {
  CreatedAt     *string `json:"created_at"`
  EnvironmentId *string `json:"environment_id"`
  Id            *string `json:"id"`
  Value         *string `json:"value"`
  VariableId    *string `json:"variable_id"`
}
